### PR TITLE
Multi-bucket Deployment

### PR DESCRIPTION
**Pull request by linl33**
## What is included in this PR?

1. This PR adds `s3_website.yml` configuration for odk2-docs and shared-docs. Given that there will be 3 `s3_website.yml`, the 3 files are placed in their respective `*-config` directory. 

2. `objects.inv` is removed from `s3_website.yml` exclude list. This file is essential for [intersphinx](http://www.sphinx-doc.org/en/master/ext/intersphinx.html).

3. CircleCI `deploy` job is split into 3: `deploy-odk1`, `deploy-odk2`, and `deploy-shared` to support separate deployment for these 3 Sphinx sites that will act as 1 website using AWS CloudFront. At this point there is no need to deploy shared, however by setting this up it will allow the cross referenced files to be added back into our documentation instead of having duplicate files copied into each deployment.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -159,8 +157,49 @@ jobs:
           name: Push to S3
           command: |
               if [[ "$CIRCLE_PROJECT_USERNAME" == "opendatakit" ]]; then \
-                s3_website push
+                s3_website push --config-dir='odk1-config'
```
Comment (yanokwa) [inline .circleci/config.yml]: Nitpicky, but you don't need quotes here.

File: .circleci/config.yml [hunk 2]
```diff
@@ -159,8 +157,49 @@ jobs:
...
+          command: |
+              if [[ "$CIRCLE_PROJECT_USERNAME" == "opendatakit" ]]; then \
+                sudo apt-get install -y openjdk-8-jre-headless
+                gem install s3_website && s3_website install
+              fi
+      - run:
+          name: Push to S3
+          command: |
+              if [[ "$CIRCLE_PROJECT_USERNAME" == "opendatakit" ]]; then \
+                s3_website push --config-dir='odk2-config'
```
Comment (yanokwa) [inline .circleci/config.yml]: No quotes needed

File: .circleci/config.yml [hunk 3]
```diff
@@ -159,8 +157,49 @@ jobs:
...
+          command: |
+              if [[ "$CIRCLE_PROJECT_USERNAME" == "opendatakit" ]]; then \
+                sudo apt-get install -y openjdk-8-jre-headless
+                gem install s3_website && s3_website install
+              fi
+      - run:
+          name: Push to S3
+          command: |
+              if [[ "$CIRCLE_PROJECT_USERNAME" == "opendatakit" ]]; then \
+                s3_website push --config-dir='shared-config'
```
Comment (yanokwa) [inline .circleci/config.yml]: No quotes

File: shared-config/s3_website.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+max_age: 300
+gzip: true
+
+exclude_from_upload:
+  - .buildinfo
+  - .doctrees
+  - .DS_Store
+  - spelling
+  - latex
```
Comment (yanokwa) [inline shared-config/s3_website.yml]: Nitpick, but I doubt we'll be building a PDF of the shared docs.

**Comment by linl33** [comment]
@yanokwa Removed those

**Pull request merged**